core/commands/refs.go

```diff
@@ -86,6 +89,12 @@ Note: List all referencess recursively by using the flag '-r'.
 			return
 		}
 
+		cached, _, err := req.Option("cached").Bool()
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+
```

Comment:
Oh wow. thank for reminding me that I hadnt merged https://github.com/ipfs/go-ipfs/pull/2257 yet.

That PR will make all this much easier